lib/rouge/formatter.rb

```diff
@@ -42,8 +42,8 @@ def self.disable_escape!
     end
 
     # Format a token stream.  Delegates to {#format}.
-    def self.format(tokens, *a, &b)
-      new(*a).format(tokens, &b)
+    def self.format(tokens, *args, **kwargs, &b)
```

Comment:
Not sure what pattern is this. It does seem like `opts` is actually a Hash in some cases?

```
html_inline.rb:      def initialize(theme)
html_legacy.rb:      def initialize(opts={})
html_line_table.rb:      def initialize(formatter, opts={})
html_linewise.rb:      def initialize(formatter, opts={})
html_pygments.rb:      def initialize(inner, css_class='codehilite')
html_table.rb:      def initialize(inner, opts={})
null.rb:      def initialize(*)
terminal256.rb:      def initialize(theme = Themes::ThankfulEyes.new)
terminal256.rb:        def initialize(style)
terminal256.rb:        def initialize(*) end
tex.rb:      def initialize(opts={})
```

Reply:
Thanks for the heads-up!

The arguments passed to `.format` should be the same ones passed to the initializer of the class - which is left completely free and is not always an options hash. Unlike lexers, formatters are solidified in code - and don't need fancy CGI-like parsing. In any case, this method should probably just blindly pass through everything - positional, keywords, and block arguments - to the constructor.

Reply:
@jneen Good to hear from you! Ok, I kept going back and forth because I wasn't sure what the actual intent. Thanks for the response! I hope you are well.